src/tests/userDetails.test.js

```diff
@@ -0,0 +1,298 @@
+    token = res.body.token;
+  });
+
+  after(async function () {
+    await UserDetailsModel.destroy({ where: { userId: user.id } });
+    await User.destroy({ where: { id: user.id } });
+  });
+  describe('POST /users/profile/:id', function () {
+    it('should create new user details when valid data is provided', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: 'Male',
+          currency: 'USD',
+          lang: 'en',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+          postalCode: '0000',
+          accountNumber: '1234',
+          accountName: 'Equity',
+          country: 'Rwanda',
+          streetAdress: 'kn109',
+        });
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property('message', 'Successfully updated');
+      expect(res.body).to.be.a('object');
+    });
+    it('should update user details that previously added)', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: 'Male',
+          currency: 'USD',
+          lang: 'en',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+          postalCode: '44444',
+          accountNumber: '1234',
+          accountName: 'Equity test',
+          country: 'Rwanda test',
+          streetAdress: 'test',
+        });
+
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property('message', 'Successfully updated');
+      expect(res.body).to.be.a('object');
+    });
+
+    it('should update user details if some optional data are not there)', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: 'Male',
+          currency: 'USD',
+          lang: 'en',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+        });
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property('message', 'Successfully updated');
+    });
+
+    it('should return 406 Bad Request, when updating user details with null params ', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: '',
+          currency: '',
+          lang: '',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+          postalCode: '0000',
+          accountName: 'Equity',
+          country: 'Rwanda',
+          streetAdress: '',
```

Comment:
Same case here. code readability.

Reply:
prob here, if you have checked well, am not using the same object properties, some where , am using them as null, just for the sake of testing. so, no need for the variable as it is used once